### ISSUE TITLE
Types of `generate` function don't match the docs

### ISSUE THREAD

**Issue by sapkra** [shellscape/webpack-manifest-plugin#276]
- Manifest Plugin Version: 4.0.0
- Webpack Version: 5
- Operating System (or Browser): macOS 11.5.2
- Node Version: 16
- Link to reproduction: no reproduction necessary

### Expected Behavior

It should be possible to return a nested javascript Object in the generate function, as documented:

> A custom Function to create the manifest. The passed function should match the signature of (seed: Object, files: FileDescriptor[], entries: string[]) => Object and can return anything as long as it's serialisable by JSON.stringify.

### Actual Behavior

One accepts an Object with string values.

### Additional Information

https://github.com/shellscape/webpack-manifest-plugin/blob/77eca2545b70ea235fc2c6b5ddf0f90540dd8588/src/index.ts#L13

https://github.com/shellscape/webpack-manifest-plugin/blob/77eca2545b70ea235fc2c6b5ddf0f90540dd8588/src/index.ts#L20-L24

The code doesn't match the docs. I think the types are wrong but prove me wrong.

**Comment by shellscape** [comment]
Issue templates are required here. Please fill one out and update your issue, and we'll reopen.

**Comment by sapkra** [comment]
@shellscape I created this issue by referencing code, so there was no hint regarding templates. I updated it now.

**Comment by shellscape** [comment]
Thanks. Will take a look as time allows.

**Comment by KutnerUri** [comment]
ah yes, it's also confusing. I've put an unnecessary `JSON.stringify()` there, and then it wouldn't parse correctly.
Type should be JSON, you can use:
https://www.npmjs.com/package/@teambit/base-ui.utils.primitives

or 
```
export type Nullable = undefined | null;
export type Primitive = string | number | bigint | boolean | Nullable;
export type JSON = { [key: string]: JSON |  Primitive | (Primitive | JSON)[] };
```

**Comment by KutnerUri** [comment]
this issue is resolved by #282.  @shellscape - can you merge? :D
